temporian/core/data/sampling.py

```diff
@@ -15,54 +15,90 @@
 """A sampling."""
 
 from __future__ import annotations
+from enum import Enum
 from typing import Dict, List, NamedTuple, Optional, Tuple, TYPE_CHECKING, Union
 
-from temporian.core.data import dtype as dtype_lib
+from temporian.core.data.dtype import DType
 
 if TYPE_CHECKING:
     from temporian.core.operators.base import Operator
 
 
-IndexDtypes = Union[dtype_lib.INT32, dtype_lib.INT64, dtype_lib.STRING]
+class IndexDTypes(Enum):
```

Comment:
I'd take the chance to rename to `IndexDType` here for consistency, so that we've `DType` and `IndexDType`.